File: singlenumber2/singlenumber.java
public class Solution {
    public int singleNumber(int[] A) {
    	Map<Integer, Integer> map = new HashMap<Integer, Integer>();
      //  HashMap map<int, int>= new HashMap<int,int>();
        for (int i : A)
        {
            try
            {
              int j = map.get(i);  
              map.put(i,j+1);
            }
            catch(Exception e)
            {
                map.put(i,1); 
            }
            if(map.get(i)==3)
                map.remove(i);
        }
        int k = (Integer) map.keySet().toArray()[0];
     
      return k;
    }
}

File: singlenumber2/singlenumber.py
class Solution:
	def singleNumber(self, A):
		d = {}
		for i in A:
	#		print i
			d[i] = d.get(i,0)+1
			if(d[i]==3):
				d.pop(i)
	#	d['3'] = 5
		return d.keys()[0]

        
s = Solution()
print s.singleNumber([5,5,5,3,2,3,3])